src/main/java/fr/insalyon/creatis/gasw/plugin/executor/dirac/execution/DiracMonitor.java

```diff
@@ -315,7 +379,32 @@ protected void resume(Job job) {
                     new JobPool(job.getFileName(), job.getCommand(), job.getParameters()));
 
         } catch (DAOException ex) {
-            // do nothing
+            logger.error("[DIRAC] error resuming job " + job.getId(), ex);
+        }
+    }
+
+    protected void finaliseJob(Job job, int exitCode) {
+
+        try {
+            job.setEnd(new Date());
+            DAOFactory factory = DAOFactory.getDAOFactory();
+
+            for (JobMinorStatus minorStatus : factory.getJobMinorStatusDAO().getExecutionMinorStatus(job.getId())) {
+
+                switch (minorStatus.getStatus()) {
+
+                    case Application:
+                        job.setRunning(minorStatus.getDate());
+                        break;
+                    case Outputs:
+                        job.setUpload(minorStatus.getDate());
+                }
```

Comment:
Why handle only job.setRunning and job.setUpload and why break after Application ? Don't we want to set the date for all minor statuses ?

Reply:
I kept the same behavior as in GaswOutputParser::parseNonStdOut.
The idea is that "queued" and "download" times should have known through dirac status changes, but, without minor statuses, "running" and "upload" times are lost when there is no output file available (killed and stalled jobs).